### ISSUE TITLE
make dev_install fails for python/kserve

### ISSUE THREAD

**Issue by spolti** [kserve/kserve#3284]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
```bash
make dev_install
```

**What did you expect to happen:**
The build to get completed.

**What's the InferenceService yaml:**
[To help us debug please run `kubectl get isvc $name -n $namespace -oyaml` and paste the output]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

The error with debug enabled:
```bash
$ make dev_install                                                                                                                                                                                                   ─╯
poetry install -v --with test,dev --extras "storage"
Using virtualenv: /Users/fspolti/data/dev/virtualenvs/ocp-ai

  GroupNotFound

  Group(s) not found: dev (via --with)

  at /opt/homebrew/Cellar/poetry/1.7.1/libexec/lib/python3.12/site-packages/poetry/console/commands/group_command.py:131 in _validate_group_options
      127│                     f"--{opt}"
      128│                     for opt in sorted(invalid_options[group])
      129│                 )
      130│                 message_parts.append(f"{group} (via {opts})")
    → 131│             raise GroupNotFound(f"Group(s) not found: {', '.join(message_parts)}")
      132│ 
make: *** [dev_install] Error 1

```

**Comment by spolti** [comment]
It seems that the group `dev` is missing or needs to be removed from the install command.

**Comment by sivanantha321** [comment]
@spolti will be fixed by #3271

**Comment by sivanantha321** [comment]
fixed by #3271 

/close
